YapDatabase/YapDatabaseTransaction.m

```diff
@@ -1569,24 +1570,29 @@ - (void)enumerateCollectionsUsingBlock:(void (^)(NSString *collection, BOOL *sto
 	if (statement == NULL) return;
 	
 	YapMutationStackItem_Bool *mutation = [connection->mutationStack push]; // mutation during enumeration protection
-	BOOL stop = NO;
+	__block BOOL stop = NO;
 	
 	// SELECT DISTINCT "collection" FROM "database2";
 	
 	int const column_idx_collection = SQLITE_COLUMN_START;
 	
-	int status;
-	while ((status = sqlite3_step(statement)) == SQLITE_ROW)
-	{
+    __block int status;
+    [self whileLoopWithChunkSize:1000
```

Comment:
`whileLoopWithChunkSize` is intended to be a drop-in replacements for YDB's existing while loops.

Note: I only applied `whileLoopWithChunkSize` to (3) loops used while rebuilding database view extensions.  We could apply it in ~10 more places, but I chose to keep this narrow to treat the specific symptom.  Ideally we should only be traversing the entire contents of large collections while (re)building database view extensions.